tsl/src/chunk_copy.c

```diff
@@ -625,6 +811,197 @@ chunk_copy(Oid chunk_relid, const char *src_node, const char *dst_node, bool del
 	}
 	PG_END_TRY();
 
-	/* Cleanup and delete the catalog entry */
-	chunk_copy_cleanup(&cc);
+	/* Finish up and delete the catalog entry */
+	chunk_copy_finish(&cc);
+}
+
+static ScanTupleResult
+chunk_copy_activity_tuple_found(TupleInfo *ti, void *const data)
+{
+	ChunkCopy **cc = data;
+
+	*cc = STRUCT_FROM_SLOT(ti->slot, ti->mctx, ChunkCopy, FormData_chunk_copy_activity);
+	return SCAN_CONTINUE;
+}
+
+static ChunkCopy *
+chunk_copy_activity_get(int32 activity_id, const char *operation_id)
```

Comment:
I think we should simplify it a bit and just use the `activity_id`

Reply:
Goes back to my API signature comment. If we decide on one then we can simplify this.

Reply:
Ok, we are in agreement to only use `activity_id`, removed the other argument.